File: closest-temperatures/code.py
def closest_to_zero(n, inputs):
    if n == 0:
        return 0

    temperatures = list(map(int, inputs))
    closest = temperatures[0]

    for temp in temperatures:
        if abs(temp) < abs(closest) or (abs(temp) == abs(closest) and temp > closest):
            closest = temp

    return closest

# Lecture des données d'entrée
n = int(input())  # le nombre de températures à analyser
inputs = input().split()  # les températures sous forme de chaîne de caractères

print(closest_to_zero(n, inputs))

File: closest-temperatures/code.js
const n: number = parseInt(readline()); // the number of temperatures to analyse
var inputs: string[] = readline().split(' ');

function closestToZero(n, inputs) {
    if (n === 0) {
        return 0;
    }

    let temperatures = inputs.map(Number);
    let closest = temperatures[0];

    temperatures.forEach(temp => {
        if (Math.abs(temp) < Math.abs(closest) || 
            (Math.abs(temp) === Math.abs(closest) && temp > closest)) {
            closest = temp;
        }
    });

    return closest;
}

const result = closestToZero(n,inputs)
    // Write an answer using console.log()
    // To debug: console.error('Debug messages...');
  
console.log(result);
  
